src/lib/gundb/FeedStorage.js

```diff
@@ -980,7 +980,9 @@ export class FeedStorage {
 
         // if no item in the cache and it's some transaction
         // then getting tx item details from the wallet
-        if (!item && id.startsWith('0x')) {
+        const shouldUpdate = get(item, 'data.reason', null) || get(item, 'data.category', null)
```

Comment:
1. there's no need to read the receipt for every item without reason. it also needs to be of type TxType.RECEIVE
2. why would category be missing and how would reading the blockchain receipt fix it?
3. also there's no need to get the whole receipt again, it should be enough to call getOutBox(event)